pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -649,13 +658,42 @@ func (a *index) inRevision(obj any) bool {
 
 // All return all known workloads. Result is un-ordered
 func (a *index) All() []model.AddressInfo {
```

Comment:
Just an observation that the contract for All() is changing here. I think this is probably fine since nothing calling the ambient index should ever see raw remote workloads that I can think of, but maybe we're better off renaming this to AllVisible or something?

/cc @howardjohn in case you have thoughts